src/lib.rs

```diff
@@ -38,17 +38,28 @@ pub enum LoadingError {
     Other(#[from] anyhow::Error),
 }
 
-#[derive(Deserialize)]
-struct RustdocFormatVersion {
-    format_version: u32,
-}
-
 fn detect_rustdoc_format_version(path: &Path, file_data: &str) -> Result<u32, LoadingError> {
-    let version = serde_json::from_str::<RustdocFormatVersion>(file_data).map_err(|e| {
-        LoadingError::RustdocFormatDetection(path.display().to_string(), anyhow::Error::from(e))
-    })?;
+    // The last characters of a rustdoc file are always: ,"format_version":NUM}
+    let start = file_data
+        .rfind(",")
```

Comment:
Let's bound the look-behind to a fixed constant that's large enough to fit the expected tail ending. There's no point in going back more than that precisely-computed small handful of bytes, because the version number will never be more than 9 digits long or it won't fit in JSON.

If the order changed, we don't want `.rfind()` to look through a MB of data before it gives up.